pkg/controllers/dynakube/injection/reconciler.go

```diff
@@ -131,32 +101,54 @@ func (r *reconciler) Reconcile(ctx context.Context) error {
 	return nil
 }
 
-func (r *reconciler) removeAppInjection(ctx context.Context) (err error) {
+func (r *reconciler) unmapDynakube(ctx context.Context) {
+	if meta.FindStatusCondition(*r.dk.Conditions(), codeModulesInjectionConditionType) != nil &&
+		meta.FindStatusCondition(*r.dk.Conditions(), metaDataEnrichmentConditionType) != nil {
+		return
+	}
+
 	namespaces, err := mapper.GetNamespacesForDynakube(ctx, r.apiReader, r.dk.Name)
 	if err != nil {
-		return errors.WithMessagef(err, "failed to list namespaces for dynakube %s", r.dk.Name)
+		log.Error(err, "failed to list namespaces for dynakube", "dkName", r.dk.Name)
 	}
 
 	dkMapper := r.createDynakubeMapper(ctx)
-
 	if err := dkMapper.UnmapFromDynaKube(namespaces); err != nil {
-		log.Info("could not unmap DynaKube from namespace")
+		log.Error(err, "could not unmap DynaKube from namespace", "dkName", r.dk.Name)
```

Comment:
very minor nitpick (and probably annoying) but here we write "DynaKube" but in the log line above we write "dynakube", maybe we should be consistent there